### PR TITLE
feat: TableCellLayout component now supports truncate prop

### LINKED ISSUE #26735: feat: Table - enable ellipsis for TableCellLayout component
Currently can be implemented like this:

Image

It would be great if the component allowed it with a prop.

### PR DESCRIPTION
**Pull request by george-cz**
## Previous Behavior

User would have to implement this feature by adding custom styles

## New Behavior

The prop `ellipsis` on `TableCellLayout` enables the ellipsis

image

## Related Issue(s)

* Fixes #26735

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/Avatar-mount_base.html">1492</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/Avatar-mount.html">1512</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/Button-mount_base.html">1087</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/Button-mount.html">1139</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProvider-mount_base.html">1870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProvider-mount.html">1863</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProviderWithTheme-mount_base.html">744</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProviderWithTheme-mount.html">729</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">685</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">685</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">737</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">738</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/InfoButton-mount_base.html">653</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/InfoButton-mount.html">668</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/MakeStyles-mount_base.html">2196</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/MakeStyles-mount.html">2180</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/Persona-mount_base.html">3566</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/Persona-mount.html">3536</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/SpinButton-mount_base.html">2913</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26738/perf-test-react-components/SpinButton-mount.html">2957</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Review by ling1726 — COMMENTED** [review]
Can you add a VR test for it in apps/vr-tests-react-components/src/stories/Table.stories.tsx

**Review by ling1726 — APPROVED** [review]

File: change/@fluentui-react-table-837a2183-555e-4693-9a8e-e3f1e89bed61.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "TableCellLayout component now supports ellipsis prop",
```
Comment (ling1726) [inline change/@fluentui-react-table-837a2183-555e-4693-9a8e-e3f1e89bed61.json:3]:
```suggestion
  "comment": "feat: TableCellLayout component now supports ellipsis prop",
```

File: packages/react-components/react-table/src/components/TableCellLayout/TableCellLayout.types.ts [hunk 2]
```diff
@@ -41,10 +41,18 @@ export type TableCellLayoutProps = ComponentProps<Partial<TableCellLayoutSlots>>
    * @default undefined
    */
   appearance?: 'primary';
+
+  /**
+   * Renders content with hidden overflow and ellipsis enabled
+   */
+  ellipsis?: boolean;
```
Comment (layershifter) [inline packages/react-components/react-table/src/components/TableCellLayout/TableCellLayout.types.ts:48]: `Text` component calls this prop `truncate`, let's be aligned

https://react.fluentui.dev/?path=/docs/components-text--default#truncate

**Pull request merged**